src/VASSAL/build/module/Chatter.java

```diff
@@ -21,477 +21,636 @@
+			@Override
       public void componentResized(ComponentEvent e) {
-        scroll.getVerticalScrollBar().setValue(scroll.getVerticalScrollBar().getMaximum());
-      }
-    });
-    input = new JTextField(80);
-    input.setFocusTraversalKeysEnabled(false);
-    input.addActionListener(new ActionListener() {
-      @Override
+				scroll.getVerticalScrollBar().setValue(scroll.getVerticalScrollBar().getMaximum());
+			}
+		});
+
+		input = new JTextField(60);
+		input.setFocusTraversalKeysEnabled(false);
+		input.addActionListener(new ActionListener() {
+			@Override
       public void actionPerformed(ActionEvent e) {
-        send(formatChat(e.getActionCommand()));
-        input.setText(""); //$NON-NLS-1$
-      }
-    });
-    input.setMaximumSize(new Dimension(input.getMaximumSize().width,
-                                       input.getPreferredSize().height));
-    scroll.setViewportView(conversation);
-    add(scroll);
-    add(input);
-  }
+				send(formatChat(e.getActionCommand()));
+				input.setText(""); //$NON-NLS-1$
+			}
+		});
+		input.setMaximumSize(new Dimension(input.getMaximumSize().width, input.getPreferredSize().height));
 
-  private String formatChat(String text) {
-    final String id = GlobalOptions.getInstance().getPlayerId();
-    return "<" + (id.length() == 0 ? "("+getAnonymousUserName()+")" : id) + "> - " + text; //$NON-NLS-1$ //$NON-NLS-2$
-  }
+		conversation.setPreferredSize(new Dimension(input.getMaximumSize().width, input.getPreferredSize().height * 11));
 
-  public JTextField getInputField() {
-    return input;
-  }
-
-  /**
-   * Display a message in the text area
-   */
-  public void show(String s) {
-    conversation.append("\n" + s); //$NON-NLS-1$
-  }
-
-  /** @deprecated use GlobalOptions.getPlayerId() */
-  @Deprecated public void setHandle(String s) {
-  }
+		scroll.setViewportView(conversation);
+		scroll.getVerticalScrollBar().setUnitIncrement(input.getPreferredSize().height); // BR// Scroll this faster																									
+		add(scroll);
+		add(input);
+	}
 
-  /** @deprecated use GlobalOptions.getPlayerId() */
-  @Deprecated public String getHandle() {
-    return GlobalOptions.getInstance().getPlayerId();
-  }
+	private String formatChat(String text) {
+    final String id = GlobalOptions.getInstance().getPlayerId();
+    return "&lt;" + (id.length() == 0 ? "(" + getAnonymousUserName() + ")" : id) + "&gt; - " + text; //$NON-NLS-1$ //$NON-NLS-2$ //BR//HTML-friendly angle brackets
+	}
 
-  /**
-   * Set the Font used by the text area
-   */
-  @Override
+	public JTextField getInputField() {
+    return input;
+	}
+	
+	//BR// To make the player colors easy to override in a custom class 
+	//BR// (my modules have logic to assign individual player colors -- beyond the scope of the present effort but a perhaps a fun future addition)
+	protected String getChatStyle(String s)
+	{	  
+    if (s.startsWith(formatChat("").trim())) { //$NON-NLS-1$
```

Comment:
Indentation looks a bit wild around this part, also there seems to be some trailing whitespace.

Reply:
Do you have a good tool suggestion to clean up all the #$%^ tabs/whitespace. It has been an ongoing disaster :)

Reply:
Ran through a whitespace prettifier I had lying around. Hopefully better on whitespace and tab crud all around.

Reply:
Code style settings in the IDE should be able to take care of that, in Eclipse under Window > Preferences > Java > Code Style > Formatter, once they are properly set they can be applied to the highlighted region or to a whole file with IIRC `ctrl+shift+f`. Took me a couple PRs to adjust my local settings to the projects standards, I'm hoping to get to a point where we have a standard code style that the developers can simply import into their IDE,  #26 is a step in that direction.